### PR TITLE
Added Objective C stress tests

### PR DESCRIPTION
**Pull request by rmstar**

### CODE REVIEW

File: src/objective-c/tests/MacTests/StressTestsCleartext.m [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+- (int32_t)encodingOverhead {
+  return kLocalInteropServerOverhead;  // bytes
+}
+
+- (void)setUp {
+  [super setUp];
+
+  // Register test server as non-SSL.
+  [GRPCCall useInsecureConnectionsForHost:[[self class] host]];
```
Comment (muxi) [inline src/objective-c/tests/MacTests/StressTestsCleartext.m]: This is old interface; can be discarded.

File: src/objective-c/tests/MacTests/StressTests.m [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
++ (void)setUp {
+  setenv(kCFStreamVarName, "1", 1);
+}
+
+- (void)setUp {
+  self.continueAfterFailure = NO;
+
+  [GRPCCall resetHostSettings];
+  _service = [[self class] host] ? [RMTTestService serviceWithHost:[[self class] host]] : nil;
```
Comment (muxi) [inline src/objective-c/tests/MacTests/StressTests.m]: Consider using v2 api `serviceWithHost:CallOptions:`

File: src/objective-c/tests/InteropTests.m [hunk 3]
```diff
@@ -341,6 +341,90 @@ - (void)testLargeUnaryRPCWithV2API {
   [self waitForExpectationsWithTimeout:TEST_TIMEOUT handler:nil];
 }
 
+- (void)testConcurrentRPCsWithErrorsWithV2API {
```
Comment (muxi) [inline src/objective-c/tests/InteropTests.m]: As discussed offline, this test is a valid interop test case, but not really putting much load on the system. Maybe have a test like this but adds more load, run longer, and run separately from the interop test suite.

File: src/objective-c/tests/MacTests/StressTestsCleartext.m (hunk 1 above)
Comment (muxi) [reply src/objective-c/tests/MacTests/StressTestsCleartext.m]: oh never mind. if you are testing v1 api too then you do need it

File: src/objective-c/tests/MacTests/StressTests.m (hunk 2 above)
Comment (rmstar) [reply src/objective-c/tests/MacTests/StressTests.m]: Fixed

File: src/objective-c/tests/InteropTests.m (hunk 3 above)
Comment (rmstar) [reply src/objective-c/tests/InteropTests.m]: I'll add a new stress test job for long running tests as part of a new PR. The new job will run periodically (not on every PR).

**Pull request merged**